elixir/apps/api/test/api/controllers/example_controller_test.exs

```diff
@@ -4,7 +4,7 @@ defmodule API.ExampleControllerTest do
   describe "echo/2" do
     test "returns error when not authorized", %{conn: conn} do
       conn = post(conn, "/v1/echo", %{"message" => "Hello, world!"})
-      assert json_response(conn, 401) == %{"error" => "invalid_access_token"}
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
```

Comment:
Would you expect `errors` to return a list or a map? From naming I would assume a list will be rendered, not a single item as a map.
```suggestion
      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
```

Reply:
I just went off of what the Phoenix generator created and figured that's what the norm was for the community.  By default here's what gets generated for the errors template:
```
defmodule MyWebApp.ErrorJSON do
  # If you want to customize a particular status code,
  # you may add your own clauses, such as:
  #
  # def render("500.json", _assigns) do
  #   %{errors: %{detail: "Internal Server Error"}}
  # end

  # By default, Phoenix returns the status message from
  # the template name. For example, "404.json" becomes
  # "Not Found".
  def render(template, _assigns) do
    %{errors: %{detail: Phoenix.Controller.status_message_from_template(template)}}
  end
end
```

I'm open to changing things, but just unsure of what type of effort would be involved if we moved away from the default.

Reply:
There should be no efforts, we can style out API responses the way we like

Reply:
Sorry, I should have been a little more clear on my last sentence.  Is there anything we miss out on by going outside the Phoenix generated template?

Reply:
Also, is it going to get strange with Ecto change set errors?  Here's an example of what currently gets returned if there are multiple errors in a changeset:
```
{
  "errors": {
    "provider_identifier": [
      "can't be blank"
    ],
    "provider_identifier_confirmation": [
      "email does not match"
    ]
  }
}
```

If we're going to make `errors` a list, would we need to have a Keyword List in some instances and a List in others?

Reply:
@bmanifold good catch with changesets. It's pretty hard to change things like this one later so maybe we can just do a minimal structuring, maybe something like this:

```
# one-off error, like unathorized
%{error: %{reason: Phoenix.Controller.status_message_from_template(template)}}
# example
%{error: %{reason: "unathorized"}}

# changesets
%{error: %{reason: "bad request", validation_errors: %{...}}}
```
or something along those lines?

This is not the most important thing though and we can also leave it as is if naming and restructuring everything will take too much time right now.

Reply:
Error responses should now be in the form recommended